pkg/sessions/redis/aws-iam/auth.go

```diff
@@ -0,0 +1,98 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	"github.com/aws/aws-sdk-go-v2/config"
+)
+
+// This code was largely copied from this repo: https://github.com/build-on-aws/aws-redis-iam-auth-golang/tree/main
+
+const (
+	// "The IAM authentication token is valid for 15 minutes"
+	// https://docs.aws.amazon.com/memorydb/latest/devguide/auth-iam.html#auth-iam-limits
+	tokenValiditySeconds = 900
+
+	connectAction = "connect"
+
+	// If the request has no payload you should use the hex encoded SHA-256 of an empty string as the payloadHash value.
+	hexEncodedSHA256EmptyString = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
+)
+
+type IAMTokenGenerator struct {
```

Comment:
Nit missing comment on exported struct

Does it actually need to be exported? Could be behind an interface?